content/security/for-admins/scim.md

```diff
@@ -58,3 +61,189 @@ You must make sure you have [configured SSO](single-sign-on/configure/_index.md)
+
+### Expression mapping
+
+This implementation works best for roles, but can't be used along with organization and team mapping using the same method. With this approach, you can assign attributes at a group level, which members can inherit. This is the recommended approach for role mapping.
+
+1. In the **Edit Attribute** view, select the **Expression** mapping type.
+2. To create app roles named as the role directly, in the **Expression** field, enter `SingleAppRoleAssignment([appRoleAssignments])`.
+3. Set the following fields:
+    - **Target attribute**: `urn:ietf:params:scim:schemas:extension:docker:2.0:User:dockerRole`.
+    - **Match objects using this attribute**: No
+    - **Apply this mapping**: Always
+4. Save your configuration.
+
+If you’re restricted to using app roles you have already defined (for example, `My Corp Administrators`) you’ll need to setup a switch for these roles. For example:
+
+```text
+Switch(SingleAppRoleAssignment([appRoleAssignments]), "member", "My Corp Administrator", "owner", "My Corp Editor", "editor")`
+```
+
+### Direct mapping
+
+This implementation works for all three mapping types at the same time.
+
+1. In the **Edit Attribute** view, select the **Direct** mapping type.
+2. Set the following fields:
+    - **Source attribute**: `extensionAttribute1`
+    - **Target attribute**: `urn:ietf:params:scim:schemas:extension:docker:2.0:User:dockerRole`
+    - **Match objects using this attribute**: No
+    - **Apply this mapping**: Always
+3. Save your configuration.
+
+### Assign users
+
+Go to **App registrations > YOUR APP > App Roles** and create an app role for each Docker role. If possible, create it with a display name that is directly equivalent to the role in Docker, for example, `owner` instead of `Owner`. If set up this way, then you can use expression mapping to `SingleAppRoleAssignment([appRoleAssignments])`. Otherwise, a custom switch will have to be used. See [Expression mapping](#expression-mapping).
+
+To add a user:
+1. Go to **YOUR APP > Users and groups**. Select **Add user/group**.
+2. Select the user you want to add, then **Select** their desired role.
+
+To add a group:
+1. Go to **YOUR APP > Users and groups**. Select **Add user/group**.
+2. Select the group you want to add, then **Select** the desired role for the users in that group.
+
+{{< /tab >}}
+{{< /tabs >}}
+
+See the documentation for your IdP for additional details:
+
+- [Okta](https://help.okta.com/en-us/Content/Topics/users-groups-profiles/usgp-add-custom-user-attributes.htm)
+- [Entra ID (formerly Azure AD)](https://learn.microsoft.com/en-us/azure/active-directory/app-provisioning/customize-application-attributes#provisioning-a-custom-extension-attribute-to-a-scim-compliant-application)
+- [OneLogin](https://onelogin.service-now.com/support?id=kb_article&sys_id=742a000d4740f1909d8dfd1f536d435f&kb_category=566ffd6887332910695f0f66cebb3556#config-info-custom)
```

Comment:
I guess we need to remove this OneLogin reference too

Reply:
Yeah can remove to avoid confusion around why we don't have steps for OL.